File: ast/src/testing/go/anonymous_functions.go
package main

// @ast node: Endpoint "/anon-get" [verb=GET]
// @ast edge: Handler -> Function "GET_anon-get_func_L16" "anonymous_functions.go"
// @ast node: Endpoint "/anon-post" [verb=POST]
// @ast edge: Handler -> Function "POST_anon-post_func_L21" "anonymous_functions.go"
// @ast node: Function "mainAnon"
// @ast node: Function "GET_anon-get_func_L16"
// @ast node: Function "POST_anon-post_func_L21"

import "github.com/gin-gonic/gin"

func mainAnon() {
	r := gin.Default()

    // Test 1: GET with anonymous func
	r.GET("/anon-get", func(c *gin.Context) {
		c.JSON(200, gin.H{"message": "anon get"})
	})

    // Test 2: POST with anonymous func
	r.POST("/anon-post", func(c *gin.Context) {
		c.JSON(200, gin.H{"message": "anon post"})
	})
}

File: ast/src/testing/php/anonymous_functions.php
<?php
// @ast node: Endpoint "/posts-closure/{id}/like" [verb=POST]
// @ast edge: Handler -> Function "post_posts_closure_{id}_like_handler_L21" "anonymous_functions.php"
// @ast node: Endpoint "/status-arrow" [verb=GET]
// @ast edge: Handler -> Function "get_status_arrow_handler_L26" "anonymous_functions.php"
// @ast node: Endpoint "/users-closure" [verb=GET]
// @ast edge: Handler -> Function "get_users_closure_handler_L17" "anonymous_functions.php"
// @ast node: Function "get_status_arrow_handler_L26"
// @ast node: Function "get_users_closure_handler_L17"
// @ast node: Function "post_posts_closure_{id}_like_handler_L21"
// @ast node: Import "import-imports-srctestingphpanonymousfunctionsphp-12"

use Illuminate\Support\Facades\Route;
use App\Models\User;
use App\Models\Post;

// Laravel closure routes
Route::get('/users-closure', function() {
    return User::all();
});

Route::post('/posts-closure/{id}/like', function($id) {
    return Post::find($id)->like();
});

// Arrow function routes
Route::get('/status-arrow', fn() => ['status' => 'ok']);
